### PR TITLE
Uml update

### LINKED ISSUE #88: Add UML Diagram

### PR DESCRIPTION
**Pull request by maahir-garg**
Fixes #88

### CODE REVIEW

File: docs/diagrams/Commands.puml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+Command <|-- FindCommand
+Command <|-- AddStudentCommand
+Command <|-- AddTeacherCommand
+Command <|-- ListCommand
+Command <|-- SortCommand
+Command <|-- DeleteCommand
+Command <|-- EditCommand
+
+' Exception handling
+exceptions.Exception <|-- exceptions.CommandException
```
Comment (Arnaxx54) [inline docs/diagrams/Commands.puml:127]: Smart to put exceptions in a separate box!

**Review by Arnaxx54 — APPROVED** [review]
LGTM

File: docs/diagrams/Commands.puml [hunk 2]
```diff
@@ -0,0 +1,128 @@
+@startuml
+!include style.puml
+skinparam arrowColor MODEL_COLOR
+skinparam classBackgroundColor MODEL_COLOR
+
+class ClearCommand {
+    + {static} String COMMAND_WORD
```
Comment (amoschee) [inline docs/diagrams/Commands.puml:7]: Good to have attributes labelled in case we include style.puml

File: docs/diagrams/Commands.puml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+Command <|-- UndoCommand
+Command <|-- HelpCommand
+Command <|-- ExitCommand
+Command <|-- FindCommand
+Command <|-- AddStudentCommand
+Command <|-- AddTeacherCommand
+Command <|-- ListCommand
+Command <|-- SortCommand
+Command <|-- DeleteCommand
+Command <|-- EditCommand
```
Comment (amoschee) [inline docs/diagrams/Commands.puml:124]: Good use of arrows to show inheritance relationships

File: docs/diagrams/Commands.puml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+
+class EditCommand {
+    + {static} String COMMAND_WORD
+    + {static} String MESSAGE_USAGE
+    + {static} String MESSAGE_EDIT_PERSON_SUCCESS
+    + {static} String MESSAGE_NOT_EDITED
+    + {static} String MESSAGE_DUPLICATE_PERSON
+    + CommandResult executeCommand(Model)
+    - {static} Person createEditedPerson(Person, EditPersonDescriptor)
```
Comment (Arnaxx54) [inline docs/diagrams/Commands.puml:108]: Good use of + and - to clearly indicate whether they are public or private.

**Review by Arnaxx54 — APPROVED** [review]

**Pull request merged**